metricbeat/module/kibana/stats/stats.go

```diff
@@ -67,46 +67,70 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 		return nil, err
 	}
 
-	statsHTTP, err := helper.NewHTTP(base)
+	return &MetricSet{
+		MetricSet: ms,
+	}, nil
+}
+
+// Fetch methods implements the data gathering and data conversion to the right format
+// It returns the event which is then forward to the output. In case of an error, a
+// descriptive error must be returned.
+func (m *MetricSet) Fetch(r mb.ReporterV2) error {
+	err := m.init()
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	kibanaVersion, err := kibana.GetVersion(statsHTTP, statsPath)
+	now := time.Now()
+
+	err = m.fetchStats(r, now)
 	if err != nil {
-		return nil, err
+		if m.XPackEnabled {
+			m.Logger().Error(err)
```

Comment:
@ycombinator Do you really need to have a special handling of erros when xpack is enabled? My understanding is that the error will be returned by the `Fetch()` method and will be logged upstream?

Reply:
Yes, this change was specifically made in https://github.com/elastic/beats/pull/12265.

In fact, I'm glad you brought this up — I need to add this same special handling around https://github.com/elastic/beats/pull/15270/files#diff-8710596f0decaae3ca552cedea1eb0a0R81 too!

Reply:
Oh good story about the different indices :)